pkg/messages/baseMessage.go

```diff
@@ -8,11 +12,13 @@ type Message interface {
 type baseMessage struct {
 	SenderFloor  int
 	responseType Message
+	messageID    string
 }
 
 func NewBaseMessage(SenderFloor int) *baseMessage {
 	msg := &baseMessage{
 		SenderFloor: SenderFloor,
+		messageID:   uuid.New().String(),
```

Comment:
The code works fine. Im not too sure how useful having a unique message ID would be. A completely new message ID doesn't act as a conversation ID by which you can 'remember' conversations and who you have talked to. Imo it also doesn't help to remember states about any other agent since it is not associated with an agent and only associated with a message. 

So I personally don't see much use in this, but what does everyone else think?